core/src/main/scala/scala/pickling/generator/ast.scala

```diff
@@ -0,0 +1,202 @@
+package scala.pickling
+package generator
+
+
+
+// 1. Read symbols for type
+// 2. Use an algorithm to determine the behavior of the pickle/unpickle methods (i..e generate an AST)
+//     CallConstructor(V)
+//     SetField(<date>)
+//     SetField(<id>)
+//     ---------------
+//     GetField(<date>)
+//     GetField(<id>)
+// 3. Generate code using IrAst
+
+
+/**
+ * An AST representing simple pickling behavior.  We generate this before we generate the implementing code.
+ *
+ *
+ */
+sealed trait IrAst {
```

Comment:
What does "Ir" and "Ast" and for? If it's Intermediate Representation Abstract Syntax Tree I'd say most ASTs are IRs with exception of Lisp. Instead of calling it AST, why not name this mini language and append `Tree`, like `PickleCommandTree`?